src/actions/userAuthActions.ts

```diff
@@ -0,0 +1,60 @@
+import { firebaseApp, firestore } from '../firebase'
+import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword, GoogleAuthProvider, signInWithPopup, onAuthStateChanged } from "firebase/auth";
+import { doc, setDoc } from "firebase/firestore";
+import {TEST_ORGANIZATION} from "../types/organization";
+import {TEST_GROUP} from "../types/group";
+
+const auth = getAuth(firebaseApp);
+
+export const signUp = (userCredentials: any) => {
```

Comment:
Please defined types for userCredentials, maybe ```{email: string, password: string, user: {id: number}}``` or something based on what you have below